core/metacling/src/TCling.cxx

```diff
@@ -4974,6 +4994,7 @@ int TCling::ReadRootmapFile(const char *rootmapfile, TUniqueString *uniqueString
 {
    // For "class ", "namespace ", "typedef ", "header ", "enum ", "var " respectively
    const std::map<char, unsigned int> keyLenMap = {{'c',6},{'n',10},{'t',8},{'h',7},{'e',5},{'v',4}};
+   bool hasPCM = rMapHasPCM(rootmapfile, fInterpreter);
 
    if (rootmapfile && *rootmapfile) {
```

Comment:
Why don't we extend the condition here and make it `if (rootmapfile && *rootmapfile && requiresRootMap(...))`.

Reply:
We just want to remove decls here, not everything in rootmap. So we cannot branch here.